books/kestrel/c/atc/defstruct.lisp

```diff
@@ -1427,7 +1427,7 @@
        (fixtype-arrayp-of-fixtype-array-of
         (pack fixtype '-arrayp-of- fixtype '-array-of))
        (valuep-when-fixtype-arrayp (pack 'valuep-when- fixtype '-arrayp))
-       (true-listp-when-fixtype-listp (pack 'true-listp-when- elem-type-listp))
+       (true-listp-when-fixtype-listp (pack 'true-listp-when- elem-type-listp '-compound-recognizer))
```

Comment:
I would also change the name of the variable to include `-compound-recognizer` or a slight abbreviation, so we can use `true-listp-when-fixtype-listp` for the variable holding the name of the rewrite rule later, but we can always do this later.